packages/core/database/lib/query/helpers/populate/apply.js

```diff
@@ -552,7 +561,7 @@ const morphToOne = async (input, ctx) => {
     const qb = db.entityManager.createQueryBuilder(type);
 
     const rows = await qb
-      .init(populateValue)
+      .init(on && type in on ? on[type] : typePopulate)
```

Comment:
I think we can now use this (to be checked with node 14 though I always have a doubt) 
`on?.[type] ?? typePopulate`

Reply:
Looks like it work as it should
![image](https://user-images.githubusercontent.com/25851739/203006265-1676fd21-4708-4b31-aa77-030c207a0ffb.png)